complex-types.d.ts

```diff
@@ -0,0 +1,19 @@
+import type {Node, Parent} from 'unist'
+
+export type NodeOfTree<
```

Comment:
```suggestion
/**
 * Internal utility to collect all descendants of in `Tree`.
 * @see https://github.com/syntax-tree/unist-util-visit-parents/blob/18d36ad/complex-types.d.ts#L43
 */
export type InclusiveDescendant<
```

Perhaps good to reference where it comes from?

And, what do you think of that original name, `InclusiveDescendant`?

Reply:
I though `NodeOfTree` was more explicit, but I can change it

Reply:
I understand.
Several terms in unist and friends are also used in other places that deal with trees, including the DOM for example. Descendant and inclusive descendant stem from that.
Some more common terms: https://github.com/syntax-tree/unist#glossary